components/Shimmer/src/Shimmer/Shimmer.cs

```diff
@@ -0,0 +1,228 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Numerics;
+using Windows.UI;
+#if WINAPPSDK
+using Microsoft.UI;
+using CommunityToolkit.WinUI.UI;
+using CommunityToolkit.WinUI.UI.Animations.Expressions;
+using CommunityToolkit.WinUI.UI.Animations;
+using Microsoft.UI.Composition;
+using Microsoft.UI.Xaml.Hosting;
+using Microsoft.UI.Xaml.Shapes;
+#else
+using Windows.UI.Composition;
+using Windows.UI.Xaml.Hosting;
+using Windows.UI.Xaml.Shapes;
+using Microsoft.Toolkit.Uwp.UI;
+using Microsoft.Toolkit.Uwp.UI.Animations.Expressions;
+using Microsoft.Toolkit.Uwp.UI.Animations;
+#endif
+
+namespace CommunityToolkit.Labs.WinUI;
+
+/// <summary>
+/// A generic shimmer control that can be used to construct a beautiful loading effect.
+/// </summary>
+[TemplatePart(Name = PART_Shape, Type = typeof(Rectangle))]
+#pragma warning disable CA1001 // Types that own disposable fields should be disposable
+public partial class Shimmer : Control
+#pragma warning restore CA1001 // Types that own disposable fields should be disposable
+{
+    private const float InitialStartPointX = -7.92f;
+    private const string PART_Shape = "Shape";
+
+    private Vector2Node _sizeAnimation;
+    private Vector2KeyFrameAnimation _gradientStartPointAnimation;
+    private Vector2KeyFrameAnimation _gradientEndPointAnimation;
+    private CompositionColorGradientStop _gradientStop1;
+    private CompositionColorGradientStop _gradientStop2;
+    private CompositionColorGradientStop _gradientStop3;
+    private CompositionColorGradientStop _gradientStop4;
+    private CompositionRoundedRectangleGeometry _rectangleGeometry;
+    private ShapeVisual _shapeVisual;
+    private CompositionLinearGradientBrush _shimmerMaskGradient;
+    private Border? _shape;
+
+    private bool _initialized;
+    private bool _animationStarted;
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    public Shimmer()
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
```

Comment:
Bringing over comment from other PR, that we should investigate clearing up this warning by using nullables.

Reply:
I made those nullable and used https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving as I wanted to avoid having a bunch of null checks.. but not sure if that's the right approach? Let me know!